### PR TITLE
Automation/device properties

### PR DESCRIPTION
**Pull request by amar-c8y**
Automation added for device properties:- Last measurement and child Assets/Devices.
Successfully run E2E :- https://github.com/SoftwareAG/cumulocity-asset-properties-widget/actions/runs/8415689363

### CODE REVIEW

File: cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts [hunk 1]
```diff
@@ -285,88 +264,409 @@ describe("Device properties widget", function () {
...
+    // User should be able to select multiple data points at once.Once selected add data point button should be enabled.
+    // User must be able to see multiple data fragments after selecting a device on data selector window.
+    it('TC_Device_Property_Measurement_015-2', () => {
+      cy.get(asset_properties_widget_elements.addPropertyButton).click();
+      cy.selectProperty(last_measurement_str);
+      cy.get(asset_properties_widget_elements.selectButton).click();
+      chooseDeviceForDataPointConfiguration();
+      cy.get('c8y-ui-empty-state')
+        .should('contain.text', 'No data points to display.')
+        .and('contain.text', 'No data points selected.');
```
Comment (dchc01) [inline cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts:474]: I believe you may encounter a lint error due to this chaining. Please check once.

File: cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts [hunk 2]
```diff
@@ -285,88 +264,409 @@ describe("Device properties widget", function () {
...
+      cy.get(asset_properties_widget_elements.dataPointAddButton).should('be.visible');
+      cy.get(`[title="${flowDataPointLabel}"]`).should('contain.text', flowDataPointLabel);
+      cy.get(asset_properties_widget_elements.addDataPointsButton).should('be.disabled');
+      cy.addOrRmoveDataPoint(flowDataPointLabel, 'add');
+      cy.addOrRmoveDataPoint(temperatureDataPointLabel, 'add');
+      cy.get(asset_properties_widget_elements.addDataPointsButton).should('be.enabled');
+      cy.get('[title="Selected data points"]')
+        .parent('div')
+        .should('contain.text', flowDataPointLabel)
+        .and('contain.text', temperatureDataPointLabel);
```
Comment (dchc01) [inline cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts:485]: The same comment applies here too.

**Review by dchc01 — CHANGES_REQUESTED** [review]

File: cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts (hunk 1 above)
Comment (amar-c8y) [reply cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts:474]: Will handle it once Cypress lint is configured.

File: cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts (hunk 2 above)
Comment (amar-c8y) [reply cypress-E2E-tests/cypress/integration/device_properties_widget.cy.ts:485]: Will handle it once Cypress lint is configured.

File: cypress-E2E-tests/cypress/support/device.commands.ts [hunk 3]
```diff
@@ -80,158 +88,183 @@ declare global {
...
+      failOnStatusCode: false
     })
-      .its("status")
-      .should("eq", 201);
+      .its('status')
+      .should('eq', 201);
+  });
+});
+
+Cypress.Commands.add('addAssetToDevice', (parentDevice, asset) => {
```
Comment (ghost) [inline cypress-E2E-tests/cypress/support/device.commands.ts:247]: In what scenario does a device have an asset as a child?

**Review by ghost — APPROVED** [review]

File: cypress-E2E-tests/cypress/support/device.commands.ts (hunk 3 above)
Comment (amar-c8y) [reply cypress-E2E-tests/cypress/support/device.commands.ts:247]: No, currently we are able to add child assets via the API.

**Review by dchc01 — APPROVED** [review]

**Pull request merged**